src/jit/lower.cpp

```diff
@@ -2321,10 +2323,14 @@ void Lowering::InsertPInvokeMethodProlog()
     store->gtOp.gtOp1 = call;
     store->gtFlags |= GTF_VAR_DEF;
 
-    GenTree* insertionPoint = m_blockRange.FirstNonPhiOrCatchArgNode();
+    GenTree* insertionPoint = firstBlockRange.FirstNonPhiOrCatchArgNode();
+    if (insertionPoint == nullptr)
+    {
+        insertionPoint = firstBlockRange.EndExclusive();
```

Comment:
I'm not liking this name very much anymore. Seems like we should have First == first node, Last == last node, Begin and End should only be on iterators.